### PR TITLE
Keyboard focus  not going to "Pause Execution" button 

### PR DESCRIPTION
**Pull request by GouriKumari**
Fixes: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/753497

### CODE REVIEW

**Comment by GouriKumari** [comment]
@netonjm : Can you take a look at this bug?  I checked a few options 
1. Added nextKeyView in https://github.com/mono/monodevelop/blob/master/main/src/addins/MacPlatform/MainToolbar/MainToolbar.cs#L104 (which wasn't the right option)
2. checked adding the changes here, https://github.com/mono/monodevelop/blob/master/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.MainToolbar/MainToolbarController.cs#L811. 
It doesn't look like the right solution .

**Comment by GouriKumari** [comment]
@sevoku: Thank you. Applied the patch and it  is working.
https://www.dropbox.com/s/kbf8yj9sfdgaiaj/Debug%20Focus.mov?dl=0

File: main/src/addins/MacPlatform/MainToolbar/ButtonBarContainer.cs [hunk 1]
```diff
@@ -61,21 +62,25 @@ public class ButtonBarContainer : NSView
...
-				return; 
-            }
-
-			if (theEvent.Characters == " " && NextKeyView != null) {
-				var buttonBar = buttonBars.FirstOrDefault ();
-				buttonBar.ExecuteFocused ();
+		{
+			if (theEvent.KeyCode == (ushort)KeyCodes.Tab) {
+				if (theEvent.ModifierFlags == (NSEventModifierMask)KeyModifierFlag.None) {
+					var success = buttonBars.FirstOrDefault ().IncreaseFocusIndex ();
```
Comment (Therzok) [inline main/src/addins/MacPlatform/MainToolbar/ButtonBarContainer.cs]: If we're using FirstOrDefault, we need a null check, otherwise this will NRE.

Comment (Therzok) [reply main/src/addins/MacPlatform/MainToolbar/ButtonBarContainer.cs]: But I'd avoid LINQ on keypresses altogether.

**Comment by sevoku** [comment]
@monojenkins backport release-8.4

**Comment by netonjm** [comment]
CQdzPH06EZ
The SHIFT tab seems to be doing odd behaviour

**Comment by GouriKumari** [comment]
@netonjm : I am not able reproduce the issue, that you are seeing. I checked with the VSFMac build  with this PR. For me, using the SHIFT+TAB is behaving correctly, it is moving between the status bar and pause/execution button correctly.

**Comment by netonjm** [comment]
@monojenkins  rebase

**Comment by netonjm** [comment]
@monojenkins rebase

**Comment by sevoku** [comment]
@monojenkins rebase

File: main/src/addins/MacPlatform/MainToolbar/MainToolbar.cs [hunk 2]
```diff
@@ -238,6 +238,7 @@ public void RebuildToolbar (IEnumerable<ButtonBarGroup> groups)
 			}
 
 			awesomeBar.ButtonBarContainer.ButtonBars = buttonBars;
+
```
Comment (Therzok) [inline main/src/addins/MacPlatform/MainToolbar/MainToolbar.cs]: Redundant newline.

File: main/src/addins/MacPlatform/MainToolbar/ButtonBarContainer.cs [hunk 3]
```diff
@@ -61,21 +62,31 @@ public class ButtonBarContainer : NSView
...
+						var buttonBar = buttonBars[0];
+						buttonBar.ExecuteFocused ();
+					}
 			}
 
-			base.KeyDown (theEvent);
-        }
+		   	base.KeyDown (theEvent);
+		}
+
```
Comment (Therzok) [inline main/src/addins/MacPlatform/MainToolbar/ButtonBarContainer.cs]: Redundant newline.

**Comment by sevoku** [comment]
@monojenkins backport release-8.4

**Comment by GouriKumari** [comment]
@monojenkins backport release-8.4

**Pull request merged**